### PR TITLE
fix: allow for numpy 1.X backwards compatibility

### PR DESCRIPTION
**Pull request by RobPasMue**
As title says. Allowing compatibility with both NumPy major versions (for now)

### CODE REVIEW

File: pyproject.toml [hunk 1]
```diff
@@ -2,7 +2,7 @@
 build-backend = "setuptools.build_meta"
 requires = [
     "cython>=3.0.0",
-    "numpy>=2,<3",
+    "numpy>=1,<3",
```
Comment (RobPasMue) [inline pyproject.toml:5]:
```suggestion
    "numpy>=1.16.0,<3",
```

**Review by germa89 — CHANGES_REQUESTED** [review]

File: ansys/mapdl/reader/cython/_binary_reader.pyx [hunk 2]
```diff
@@ -364,7 +364,11 @@ cdef np.ndarray wrap_array(void* c_ptr, int size, int type_flag, int prec_flag):
     array_wrapper.set_data(size, c_ptr, my_dtype)
 
     cdef np.ndarray ndarray
-    ndarray = np.array(array_wrapper)
+    # TODO: Remove backwards compatibility in the future
+    if np.__version__.startswith("1."): # Backwards compatibility with numpy 1.X
+        ndarray = np.array(array_wrapper, copy=False)
```
[the remark this answers is not in the document]
Comment (akaszynski) [reply ansys/mapdl/reader/cython/_binary_reader.pyx:369]: `copy=False` is indeed for performance reasons on numpy 1.X. Looking at the docs now, seems like the ideal approach would be `copy=None` for 2.X, as it copies only when necessary.

[the remark this answers is not in the document]
Comment (RobPasMue) [reply ansys/mapdl/reader/cython/_binary_reader.pyx:369]: Sure, I can change that.

File: ansys/mapdl/reader/cython/_binary_reader.pyx [hunk 3]
```diff
@@ -364,7 +364,11 @@ cdef np.ndarray wrap_array(void* c_ptr, int size, int type_flag, int prec_flag):
...
+    else:
+        ndarray = np.array(array_wrapper)
```
Comment (RobPasMue) [inline ansys/mapdl/reader/cython/_binary_reader.pyx:371]:
```suggestion
        ndarray = np.array(array_wrapper, copy=None)
```

**Comment by RobPasMue** [comment]
@germa89 - comments solved. Review and approve.

**Review by akaszynski — APPROVED** [review]
LGTM.

**Review by germa89 — APPROVED** [review]
Looks good to me. Thank you @RobPasMue and @akaszynski :)

**Comment by RobPasMue** [comment]
Let me get the release done now, 0.54

**Pull request merged**